### DISCUSSION TITLE
Register middleware from module globally

### DISCUSSION THREAD

**Discussion by ghostzero** [nuxt/nuxt#19178] [Questions] [answered]
Hey,

Is there an easy way to glogal register a middleware from a third-party nuxt module? I want to avoid using `definePageMeta` on every page just to apply a middleware globally.

Currently there is no migration path suggested in Nuxt 3:

- https://nuxt.com/docs/migration/plugins-and-middleware
- https://nuxtjs.org/docs/directory-structure/middleware/#router-middleware

Thanks!

**Comment by Baroshem** [comment, marked as answer]
Hey,

Yes it is possible :)

Check out how I am declaring global middleware in my modules:
- https://github.com/Baroshem/nuxt-security/blob/f2a994ca12fc754d61985c7b93c1e854f6345f88/src/module.ts#L100
- https://github.com/Baroshem/nuxt-security/blob/main/src/runtime/server/middleware/rateLimiter.ts

Or in Algolia module:
- https://github.com/nuxt-modules/algolia/blob/d0be7f7eeec419fb98c22a09beb311e9c8181cb8/src/module.ts#L183
- https://github.com/nuxt-modules/algolia/blob/main/src/runtime/server/api/storyblok.ts
